### PR TITLE
[SW-2425] Remove max_categorical_levels from GBM API

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/IgnoredParameters.scala [hunk 1]
```diff
@@ -37,10 +37,13 @@ object IgnoredParameters {
...
   def common: Seq[String] = deprecated ++ implementedInParent ++ unimplemented
 
-  def all(algorithm: String): Seq[String] = algorithm match {
-    case "H2OKMeans" => common ++ Seq("response_column", "offset_column")
-    case "H2OGAM" => common ++ Seq("plug_values") // According to MK the parameter doesn't make much sense for GAM
-    case "H2ODeepLearning" => common ++ Seq("pretrained_autoencoder")
-    case _ => common
+  def all(algorithm: String): Seq[String] = common ++ {
+    algorithm match {
+      case "H2OGBM" => Seq("max_categorical_levels")
```
Comment (maurever) [inline api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/IgnoredParameters.scala:42]: Why is it removed for GBM? I think it is a valid parameter for all tree algos. I checked the GBM documentation, and the parameter is missing there; however, it is probably a bug. Or am I missing something?

Comment (mn-mikke) [reply api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/IgnoredParameters.scala:42]: I can't find this parameter in [MOJO json file](https://gist.github.com/mn-mikke/8177b9503bcffb124131c5b8e15b5b46#file-gbm-model-json) so I thought the parameter is not related to the algorithm. If think that the parameter should be there, I can close this PR, but I need help why it's not propagated to MOJO.

Comment (maurever) [reply api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/IgnoredParameters.scala:42]: Maybe the reason why it is missing in your MOJO is that this parameter is relevant only if `categorical_encoding = "EnumLimited" `.

Comment (mn-mikke) [reply api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/IgnoredParameters.scala:42]: Will try, thanks!

Comment (mn-mikke) [reply api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/IgnoredParameters.scala:42]: @maurever I've tried to set [`categorical_encoding` to `EnumLimited`](https://gist.github.com/mn-mikke/8177b9503bcffb124131c5b8e15b5b46#file-gbm-model-enumlimited-json-L1170),  but `max_categorical_levels` is still missing.

Comment (maurever) [reply api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/IgnoredParameters.scala:42]: So, it looks like a bug. Please, make a JIRA for it and asign it to me. Thank you @mn-mikke!

Comment (mn-mikke) [reply api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/IgnoredParameters.scala:42]: @maurever I've created [this JIRA ticket](https://0xdata.atlassian.net/browse/PUBDEV-7768) for the problem. Thanks a lot! Closing the PR...

**Comment by mn-mikke** [comment]
This problem will be handled in https://0xdata.atlassian.net/browse/PUBDEV-7768

**Pull request closed without merging**